ide/lsp.client/src/org/netbeans/modules/lsp/client/LSPBindings.java

```diff
@@ -400,9 +403,17 @@ public InitializeResult getInitResult() {
         return initResult;
     }
 
-    @SuppressWarnings("AccessingNonPublicFieldOfAnotherObject")
+    @SuppressWarnings({"AccessingNonPublicFieldOfAnotherObject", "NestedSynchronizedStatement"})
```

Comment:
Seems like a mistake in the hint, will try to take a look.

Reply:
You mean the hint is wrong because the synchronisation is inside a lambda, that is not executed in the containing method? My first though was, that the hint is right, but after reading twice, you are right.

With the next push, this will be gone again, as as your proposed solution is much easier and does not need explicit synchronisation.